### PR TITLE
Pattern Assembler: Improve a11y and keyboard navigation

### PR DESCRIPTION
**Pull request by arthur791004**
Related to https://github.com/Automattic/wp-calypso/issues/74188, https://github.com/Automattic/wp-calypso/issues/74189

## Proposed Changes

* Focus on the first tabbable element when you navigate to another screen
* Enable the focus restoration when you navigate back to the previous screen. It looks good after we upgraded `@wordpress/components`
* Support keyboard navigation (arrow keys) on the options of the list (e.g. Categories, Patterns, Variations)

https://user-images.githubusercontent.com/13596067/233023019-d6cc98d0-72b1-412b-9bc8-d37395c68857.mov

## Testing Instructions

* Go to /setup/site-setup?siteSlug=<your_free_site>
* Continue until you land on the Design Picker
* On the Design Picker screen, scroll down to the bottom and select BCPA CTA
* On the Pattern Assembler screen
  * Use the keyboard to select the header, sections, footer, colors, and fonts
  * Ensure it's friendly to assemble your homepage via the keyboard on the sidebar

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx [hunk 1]
```diff
@@ -557,9 +558,7 @@ const PatternAssembler = ( {
 
 				{ isEnabledColorAndFonts && (
 					<NavigatorScreen path={ NAVIGATOR_PATHS.COLOR_PALETTES }>
-						<AsyncLoad
-							require="./screen-color-palettes"
-							placeholder={ null }
+						<ScreenColorPalettes
```
Comment (arthur791004) [inline client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx:561]: Focusing on the first tabbable element doesn't work on `AsyncLoad`

File: packages/global-styles/src/components/font-pairing-variations/style.scss [hunk 2]
```diff
@@ -36,16 +36,16 @@
 		}
 	}
 
-	&:hover,
-	&:focus-visible {
+	&.is-active {
```
Comment (arthur791004) [inline packages/global-styles/src/components/font-pairing-variations/style.scss:39]: Move it up to display focus state on the active element

File: packages/global-styles/src/components/color-palette-variations/style.scss [hunk 3]
```diff
@@ -36,16 +36,16 @@
 		}
 	}
 
-	&:hover,
-	&:focus-visible {
+	&.is-active {
```
Comment (arthur791004) [inline packages/global-styles/src/components/color-palette-variations/style.scss:39]: Move it up to display focus state on the active element

**Comment by taipeicoder** [comment]
Nice work! I have a couple of observations:

1. Keyboard navigation doesn't seem to work on the main screen:
Screenshot 2023-04-19 at 5 55 07 PM

2. I see an odd red border on the Continue button's focus state:
Screenshot 2023-04-19 at 5 53 04 PM

**Comment by arthur791004** [comment]
> Keyboard navigation doesn't seem to work on the main screen:

Yes, should we also need to focus on the first focusable element on the sidebar?

> I see an odd red border on the Continue button's focus state:

Yes, it's related to https://github.com/Automattic/wp-calypso/issues/74633 and we will fix it when we work on that ticket 🙂

**Comment by taipeicoder** [comment]
> Yes, should we also need to focus on the first focusable element on the sidebar?

That sounds good to me 🙂

**Comment by arthur791004** [comment]
> That sounds good to me 🙂

Done!

> Keyboard navigation doesn't seem to work on the main screen:

Sorry, I found I might misunderstand this. What do you mean it doesn't seem to work? I think they're independent items, so I didn't apply keyboard navigation (arrow keys) on them. Does it make sense?

**Comment by taipeicoder** [comment]
> Sorry, I found I might misunderstand this. What do you mean it doesn't seem to work? I think they're independent items, so I didn't apply keyboard navigation (arrow keys) on them. Does it make sense?

Yes, I was referring to keyboard navigation (up/down) on the main screen, it seemed odd that I could use up/down in all screens except for the main one, but maybe that's just me. We can revisit this behavior if others have the same feedback 🙂

**Review by taipeicoder — APPROVED** [review]
Asides from the unit test issues, the code looks good to me 👍
TIL about [Reakit](https://reakit.io/), very cool!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7935507

Hi @arthur791004, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Blank pattern`
- `Block patterns`
- `Block pattern categories`
- `Color palette variations`
- `Font: %s`
- `Font pairing variations`

Thank you in advance!

**Review by okmttdhr — COMMENTED** [review]
Very cool to use Composite API!
Thank you for working on this PR 💚

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7935507).

**Pull request merged**